exercises/react/react_test.py

```diff
@@ -52,43 +52,55 @@ def test_compute_cells_can_depend_on_other_compute_cells(self):
     def test_compute_cells_fire_callbacks(self):
         input_ = InputCell(1)
         output = ComputeCell([input_], lambda inputs: inputs[0] + 1)
-        callback1 = Callback()
+
+        def callback1(value):
+            return value
 
         output.add_callback(callback1)
         input_.value = 3
-        self.assertEqual(callback1.values, [4])
+        self.assertEqual(output.expect_callback_values(callback1), [4])
```

Comment:
I'm not sure that I like `expect_callback_values` being a member of `ComputeCell`...

What do you think about something like this?

```Python
callback_buffer = []
output.add_callback(callback_buffer.append)
input_.value = 3
self.assertEqual(callback_buffer, [4])
```

Reply:
You mean convert `callback` to `callback_buffer`, and keep outputs of callbacks in the buffer? If so, I guess [problem specification](https://github.com/exercism/problem-specifications/tree/master/exercises/react) of this exercise should be changed, too.

My justification for `expect_callback_values` being a member of `ComputeCell` is that `ComputeCell` seems to be acting as a **host** for callbacks to be attached to and detached from. Or you can imagine callbacks as plugins for the `ComputeCell`, so trying to obtain output values of a callback is like reading the state of a plugin.

However, these are just analogies. Insights are welcome.

Reply:
There are discussions on rewording the canonical-data, but I don't think that semantic changes are necessary to it. 

> Or you can imagine callbacks as plugins for the ComputeCell, so trying to obtain output values of a callback is like reading the state of a plugin.

Close: this is more of a publisher/subscriber scenario. The `ComputerCell` "publishes" new values by calling the registered callbacks. The callback is the subscription mechanism. The publisher doesn't care what the subscriber object is or how it uses the published data, and has no access to it. In a practical use case, if one were trying to use the solution for this exercise to perform calculations, they might simply use `output.add_callback(lambda v: print(v))`

I don't remember if I linked @petertseng's comment before (https://github.com/exercism/problem-specifications/pull/1194#issuecomment-367780745), but it's a good in-context note on what a callback is/should be.